src/Razor/src/Microsoft.AspNetCore.Razor.Language/DefaultDirectiveSyntaxTreePass.cs

```diff
@@ -46,6 +46,20 @@ public RazorSyntaxTree Verify()
                 return rewrittenTree;
             }
 
+            public override SyntaxNode Visit(SyntaxNode node)
+            {
+                try
+                {
+                    return base.Visit(node);
+                }
+                catch (InsufficientExecutionStackException)
+                {
+                    // We're very close to reaching the stack limit. Let's not go any deeper.
+                    // It's okay to not show nested section errors in deeply nested cases instead of crashing.
+                    return node;
```

Comment:
I think it's far better to show an error rather than crash, and I agree in principle that putting this in the base class is the smartest thing to do.

Another idea would be to "give up" and have the parser report and error, then treat the rest of the content as plain text if it sees something so deeply nested. Then we've avoid creating data structures that are too large to traverse.

Reply:
My reservation with catching the error is that putting this in the base class seems too general. If you have a bug you want to know... if we're going to be safe by catching the stackguard exception then we need to add an error so that it doesn't seem like a successful case.

Reply:
Oh I see you meant, even though the parser wouldn't throw this, we have a set limit of nesting that we allow in the parser and stop parsing after that. I assume that could work too but we might be limiting ourselves too much